langlib/langlib-test/src/test/resources/test-src/regexp_test.bal

```diff
@@ -1311,6 +1311,17 @@ function testTranslatingDiffNodesInCharClass() {
     assertEquality("A", (<regexp:Span>res8).substring());
 }
 
+function testRegexpWithUnicodeChars() {
+    regexp:Span? resA1 = regexp:find(re = re `\u{1F30D}`, str = "😀 Hello, 🌍! ❤️");
+    assertTrue(resA1 is regexp:Span);
+    regexp:Span? resA2 = regexp:find(re = re `\u{1F31F}`, str = "😀 Hello, 🌍! ❤️");
+    assertTrue(resA2 is ());
+    regexp:Span? resA3 = regexp:find(re = re `\u{0d9c}`, str = "පරිගණක 10");
+    assertTrue(resA3 is regexp:Span);
+    regexp:Span[] resA4 = regexp:findAll(re = re `\u{1F30D}`, str = "😀 Hello, 🌍🌍! ❤️");
+    assertTrue(resA4 is regexp:Span[]);
+}
+
```

Comment:
Lets add a test with non capturing groups with `CharSetRange`.

```
re `(?i:.*[\u{0d9c}-\u{1F30D}].*)`
```

Reply:
And a invalid unicode value like

```
re `\u{1F30D?.}`
```

Reply:
And we can have a one test with `Characterset`

```
re `[\u{1F30D}\u{1F30E}]`
```

Reply:
Added in https://github.com/ballerina-platform/ballerina-lang/pull/40368/commits/e26daea0c1d5b4a0c11d068f104eda74dd5ce5c5.